File: src/main/java/com/itsci/mju/maebanjumpen/service/SkillTypeService.java
package com.itsci.mju.maebanjumpen.service;

import com.itsci.mju.maebanjumpen.dto.SkillTypeDTO;
import java.util.List;

public interface SkillTypeService {
    // เปลี่ยน return type และ parameter เป็น DTO
    List<SkillTypeDTO> getAllSkillTypes();
    SkillTypeDTO getSkillTypeById(int id);
    SkillTypeDTO saveNewSkillType(SkillTypeDTO skillTypeDto);
    SkillTypeDTO updateSkillType(int id, SkillTypeDTO skillTypeDto);
    void deleteSkillType(int id);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/skilltype/service/SkillTypeService.kt
package com.itsci.mju.maebanjumpen.skilltype.service

import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO
import com.itsci.mju.maebanjumpen.skilltype.request.CreateSkillTypeRequest
import com.itsci.mju.maebanjumpen.skilltype.request.UpdateSkillTypeRequest

interface SkillTypeService {
    fun listAllSkillTypes(): List<SkillTypeDTO>

    fun getSkillTypeById(id: Long): SkillTypeDTO

    fun createSkillType(request: CreateSkillTypeRequest): SkillTypeDTO

    fun updateSkillType(
      request: UpdateSkillTypeRequest
    ): SkillTypeDTO

    fun deleteSkillType(id: Long): Boolean
}

